src/twisted/python/deprecate.py

```diff
@@ -714,10 +787,15 @@ def function(tweedledum=1, tweedledee=2):
     @rtype: 1-argument callable taking a callable and returning a callable.
     """
     def wrapper(wrappee):
-        argspec = inspect.getargspec(wrappee)
+        if hasattr(inspect, "getfullargspec"):
+            # Python 3
+            spec = inspect.signature(wrappee)
+        else:
+            # Python 2
+            spec = inspect.getargspec(wrappee)
         @wraps(wrappee)
         def wrapped(*args, **kwargs):
-            arguments = _passed(argspec, args, kwargs)
+            arguments = _passed(spec, args, kwargs)
             for this, that in argumentPairs:
                 if this in arguments and that in arguments:
                     raise TypeError("nope")
```

Comment:
This is a fairly unfortunate exception (though it's not new in this PR so it would be better to fix it separately).